src/include/sof/audio/component_ext.h

```diff
@@ -55,6 +55,7 @@ static inline void comp_free(struct comp_dev *dev)
 	    dev->task) {
 		schedule_task_free(dev->task);
 		rfree(dev->task);
+		dev->task = NULL;
```

Comment:
This seems a bit out of place. Is this just a generic fix, or somehow related to the DP change?

Reply:
Just a fix, leaving dangling pointers is always a bad idea. 
You right, moving to separate commit